autogluon/utils/tabular/ml/models/abstract/abstract_model.py

```diff
@@ -203,7 +205,30 @@ def preprocess(self, X):
             self.features = list(X.columns)  # TODO: add fit and transform versions of preprocess instead of doing this
         return X
 
-    def fit(self, X_train, Y_train, **kwargs):
+    def _preprocess_fit_args(self, **kwargs):
+        time_limit = kwargs.get('time_limit', None)
+        max_time_limit_ratio = self.params_aux.get('max_time_limit_ratio', 1)
+        if time_limit is not None:
+            time_limit *= max_time_limit_ratio
+        max_time_limit = self.params_aux.get('max_time_limit', None)
+        if max_time_limit is not None:
+            if time_limit is None:
+                time_limit = max_time_limit
+            else:
+                time_limit = min(time_limit, max_time_limit)
+        min_time_limit = self.params_aux.get('min_time_limit', 0)
+        if min_time_limit is None:
+            time_limit = min_time_limit
```

Comment:
In documentation it says that min_time_limit is applied after max_time_limit, so min_time_limit takes precedence, this would only ever happen if max_time_limit was less than min_time_limit though, which is generally an invalid state.

Agreed regarding clarification.